addons/stock/models/stock_move_line.py

```diff
@@ -484,13 +484,17 @@ def _log_message(self, record, move, template, vals):
         record.message_post_with_view(template, values={'move': move, 'vals': dict(vals, **data)}, subtype_id=self.env.ref('mail.mt_note').id)
 
     def _free_reservation(self, product_id, location_id, quantity, lot_id=None, package_id=None, owner_id=None, ml_to_ignore=None):
+        self.ensure_one()
+        rounding = self.product_uom_id.rounding
+        self._free_reservation_impl(product_id, location_id, quantity, rounding, lot_id=lot_id, package_id=package_id, owner_id=owner_id, ml_to_ignore=ml_to_ignore)
+
+    def _free_reservation_impl(self, product_id, location_id, quantity, rounding, lot_id=None, package_id=None, owner_id=None, ml_to_ignore=None):
```

Comment:
@sle-odoo A better name? :wink:

Reply:
This refacotring makes me also question why the move's UOM rounding should be applied on the product's UOM operations.
- If you have t with a precision of 0, it doesn't matter if you round kg to 0, to 0.0 or to 0.00, in order to determine if in terms of tones, this can be considered 0
- If you have t with a precision of 0, it doen't matter if you round MT to p, 0.0, or 0.00, neither.

This might even be a bug.

Reply:
This might even be a bug, isn't it?